Mathlib/RingTheory/DedekindDomain/FiniteAdeleRing.lean

```diff
@@ -336,9 +336,31 @@ instance : Algebra R (FiniteAdeleRing R K) :=
 instance : IsScalarTower R K (FiniteAdeleRing R K) :=
   IsScalarTower.of_algebraMap_eq' rfl
 
+/-- Coercion `FiniteAdeleRing R K → K_hat R K`. -/
+@[coe]
+def coe (x : FiniteAdeleRing R K) : K_hat R K := x.1
+
 instance : Coe (FiniteAdeleRing R K) (K_hat R K) where
   coe x := x.1
 
+@[simp, norm_cast]
+theorem coe_one : (1 : FiniteAdeleRing R K) = (1 : K_hat R K) := rfl
+
+@[simp, norm_cast]
+theorem coe_zero: (0 : FiniteAdeleRing R K) = (0 : K_hat R K) := rfl
+
+@[simp, norm_cast]
+theorem coe_add (x y : FiniteAdeleRing R K) : (x + y : FiniteAdeleRing R K) =
+  (x : K_hat R K) + (y : K_hat R K) := rfl
+
+@[simp, norm_cast]
+theorem coe_mul (x y : FiniteAdeleRing R K) : (x * y : FiniteAdeleRing R K) =
+  (x : K_hat R K) * (y : K_hat R K) := rfl
+
+@[simp, norm_cast]
+theorem coe_algebraMap (x : K) : (((algebraMap K (FiniteAdeleRing R K)) x) : K_hat R K) =
+      (algebraMap K (ProdAdicCompletions R K)) x := rfl
+
```

Comment:
These lemmas are all fine, but I am a bit confused about why we want to coerce to K_hat in the first place. What is the application here? I was hoping that we'd be able to do everything in `FiniteAdeleRing R K` without ever mentioning K_hat again (I regard K_hat as an implentation detail). Is this not the case?

Reply:
These lemmas were added via your suggestion in my PR 269 of FLT; the application being that FiniteAdeleRing.baseChange was built from the ProdAdicCompletions.baseChange and these facts were needed in the proof. I guess the answer to your question depends how often we will be defining maps on the FiniteAdeleRing via maps on K_hat and taking their restriction?